src/components/SettingsModal/index.jsx

```diff
@@ -73,15 +73,30 @@ export default class SettingsModal extends React.Component {
 
     const nWorkersIsValid = validateNWorkers(this.state.localSettings.nWorkers)
 
+    // define a custom button component to have a gear icon and no backgroundp
+    const CustomButton = React.forwardRef(({ children, onClick }, ref) => (
```

Comment:
I'd love to learn about the reasoning behind using `forwardRef` here. It's the first time I'm seeing it and just started reading through the React documentation.

Reply:
My only reasoning was "I copied it from the bootstrap docs" lol. [Here](https://react-bootstrap.netlify.app/components/dropdowns/#custom-dropdown-components) is the example I used.

Reply:
I think the bootstrap `Dropdown` uses it to control whatever you render in the custom element.

Reply:
This stuff is also new to me. I've avoided Refs because I don't fully understand when they're truly called for. 

Kinda related, I wonder about the possible approaches for using icons in our app in general. I don't know much yet, but it seems like there are some libraries that provide icons as react components directly, which might reduce the need for some of this boilerplate stuff. On the other hand, it might be nice to just go straight to the source, like you've done here. Anyway, it could be worth thinking about this a bit before committing one way or another.

Here's just one example https://material-ui.com/components/icons/#material-icons